### DISCUSSION TITLE
issues on 1.0.x on iOS

### DISCUSSION THREAD

**Discussion by driff** [supabase-community/supabase-kt#198] [General]
Hi, i've been working on a kmm project and i've finally come to the part where i need to build a release version for iOS, and it is working as expected on debug but on release it gives me some issues with the decoder on iosArm64, it might have something to do with https://kotlinlang.org/docs/whatsnew1820.html#support-for-base64-encoding 

what i did for now is reverting to version 0.9.5 and also rolled back to kt-1.8.10 and coroutines 1.6.4 and it seems to be working, i might try to find a fix but for now the downgrade seems to be working.

this is the error from the release build:
e: Module "io.github.jan-tennert.supabase:gotrue-kt (io.github.jan-tennert.supabase:gotrue-kt-iosarm64)" has a reference to symbol kotlin.io.encoding/Base64.encode|-7035830172202227036[0]. Neither the module itself nor its dependencies contain such declaration.

**Comment by jan-tennert** [comment]
That is weird, and you are sure that you have Kotlin 1.8.20/1.8.22 when using 1.0.x? The Base64 class is only used in one [class](https://github.com/supabase-community/supabase-kt/blob/master/GoTrue/src/commonMain/kotlin/io/github/jan/supabase/gotrue/PKCE.kt) which was added in 1.0.0. Do you are you able to share your project or the build files? Otherwise you can also ask in the [kotlin slack](https://slack.kotlinlang.org/) as this seems more of a kotlin problem, but feel free to share your code here and I'll try my best!

**Reply by driff** [reply]
yeah, it is a kotlin problem i'm sure of it, i had to downgrade to 1.8.10 , i also did the following trying to replicate, i created a new kmm project and implemented supabase and i couldn't build a release for iOS with 1.8.20
and i just tried with 1.8.22 and i get the same error as in the empty project:

```
> Task :shared:linkReleaseFrameworkIosArm64 FAILED
error: Compilation failed: Internal compiler error: no implementation found for FUN DEFAULT_PROPERTY_ACCESSOR name:<get-parent> visibility:public modality:ABSTRACT <> ($this:kotlinx.coroutines.Job) returnType:kotlinx.coroutines.Job?
when building itable for CLASS INTERFACE name:Job modality:ABSTRACT visibility:public superTypes:[kotlin.coroutines.CoroutineContext.Element]
implementation in CLASS CLASS name:ChannelJob modality:FINAL visibility:private superTypes:[io.ktor.utils.io.ReaderJob; io.ktor.utils.io.WriterJob; kotlinx.coroutines.Job]
at /opt/buildAgent/work/49d4a482a8522285/ktor-io/common/src/io/ktor/utils/io/Coroutines.kt (156:1)
CLASS CLASS name:ChannelJob modality:FINAL visibility:private superTypes:[io.ktor.utils.io.ReaderJob; io.ktor.utils.io.WriterJob; kotlinx.coroutines.Job]
```
using 
supabase-version = "1.0.3"
coroutines-version = "1.7.1"

the versions that seem to be working are:
supabase-version = "0.9.5"
coroutines-version = "1.6.4"

i tried the following combinations:
kotlin = 1.8.10 ~ 1.8.22 -> No effect
ktor = 2.3.1 
coroutines-version = "1.6.4" works but upgrading to 1.7.x breaks the build so far and it seems related to 
this issue: https://youtrack.jetbrains.com/issue/KTOR-5728

after this findings i think i'll need to try downgrading coroutines version to 1.6.4 in supabase-kt

**Reply by jan-tennert** [reply]
I can quickly publish a fix for the ktor problem as the bug is fixed in a newer version, give me 30mins. But the error you showed me intially seems to be another problem

**Reply by jan-tennert** [reply]
can you try out `1.0.4-dev`?

**Reply by driff** [reply]
i think the issue was with the coroutines library version, i'm updating rn, be back in 5

**Reply by jan-tennert** [reply]
yea I think I fixed that problem, but for the Base64 error you might want to ask in the kotlin slack. I've never programmed on IOS before, might be some misconfiguration?

**Reply by driff** [reply]
it's building now! have to try running

**Reply by jan-tennert** [reply]
Published `1.0.4` which also includes this fix, but be sure to add this maven repository: `https://maven.pkg.jetbrains.space/public/p/ktor/eap`

**Reply by driff** [reply]
yes, thank you! already did when i tried the -dev release, i'm new to iOS too hahaha

**Reply by driff** [reply]
also if you are using compose, the compose compiler version 1.4.7 is not compatible with kotlin 1.8.22 so you'll have to use 1.8.21 but thats all.

**Reply by jan-tennert** [reply]
Does it work now or does your initial problem still appear?

**Reply by driff** [reply]
it is working good with the -dev release, thanks! I'm switching now to 1.0.4
